### PR TITLE
Cb 8 zip code api

### PR DESCRIPTION
**Pull request by Aarushi-Pandey**
* **Please check if the PR fulfills these requirements**
- [x] The pull request title contains the ticket number in the title. For example [CB-1]

* **What kind of change does this PR introduce?** (Bug fix, new feature...)
calls zipcode API using a supplied zip code

* **What is the current behavior?What does the code do before you made the new changes?** 
Currently, the application does not call the zipcode api.

* **What is the new behavior?**
The application calls the zipcode api and returns a json of the data from localhost8080.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR? Run a new command?)
Add the weather api key to the .env file

* **Other information**:

### CODE REVIEW

**Comment by Codubee** [comment]
Hey Arushi. I think there are some changes in here that are not part of the zip code api

File: backend/index.js [hunk 1]
```diff
@@ -6,12 +6,26 @@ app.use(express.json())
...
+    })
+    .catch((error) => {
+        console.log(error);
+        res.status(400).json({error: "An error occured with the zipcode API"})
+    })
 })
 
 app.listen(port,()=>{
-    console.log('API is up and running')
+    console.log('Zipcode API is up and running')
```
Comment (JerryTeng01) [inline backend/index.js]: change this since zip code api is not the only api running

File: backend/index.js [hunk 2]
```diff
@@ -6,12 +6,26 @@ app.use(express.json())
 
 const port = 8080;
 
-app.get('/',(req,res)=>{
-    res.send('Hello world');
+const zipcode = 75080;
+const distance = 50;
+
+app.get('/api/zipcodes',(req,res)=>{
+
```
Comment (JerryTeng01) [inline backend/index.js:15]: you can get the parameters from req.query instead of hardcoding them

File: backend/index.js [hunk 3]
```diff
@@ -6,12 +6,26 @@ app.use(express.json())
...
+    const headers = {headers: {authorization: process.env.API_KEY}}
```
Comment (JerryTeng01) [inline backend/index.js]: change the name of the API key to something that describes what it is since there will be multiple api keys

Comment (JerryTeng01) [reply backend/index.js]: does this API actually require a key? It doesnt seem like it does

**Review by JerryTeng01 — COMMENTED** [review]
Seems good, just make sure to include the changes from main

File: frontend/src/components/UserInput/Zipcode/Zipcode.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import React from 'react'
```
Comment (Codubee) [inline frontend/src/components/UserInput/Zipcode/Zipcode.js]: Is this file part of this pr?

File: frontend/src/components/UserInput/Zipcode/Zipcode.css [hunk 5]
```diff
@@ -0,0 +1,21 @@
+.container {
```
Comment (Codubee) [inline frontend/src/components/UserInput/Zipcode/Zipcode.css]: Is this file part of this pr?

File: frontend/src/components/UserInput/UserInput.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
```
Comment (Codubee) [inline frontend/src/components/UserInput/UserInput.js]: Is this file part of this pr?

File: frontend/src/App.js [hunk 7]
```diff
@@ -1,5 +1,6 @@
 import './App.css';
```
Comment (Codubee) [inline frontend/src/App.js]: Is this file part of this pr?

File: frontend/package.json [hunk 8]
```diff
@@ -3,6 +3,7 @@
   "version": "0.1.0",
```
Comment (Codubee) [inline frontend/package.json]: Is this file part of this pr?

**Review by Codubee — CHANGES_REQUESTED** [review]
There are some files for the frontend when I dont think there should be. Might consider removing them from this pr

**Comment by Aarushi-Pandey** [comment]
Thanks so much for the detailed feedback Miguel and Jerry! It really helped me improve my code efficiently and quickly. I've updated this branch so that there are no merge conflicts with main and have changed my hardcoded zipcode and distance values in index.js to query parameters as Jerry suggested. Great job in figuring out that there was no need for the API key for the zipcode API as well! I've removed that as well. I've attempted to remove frontend files from this pr too. Please let me know if there are further changes to be made! Thanks once more for taking the time to give me detailed comments and suggestions :)

File: backend/index.js [hunk 9]
```diff
@@ -6,6 +6,24 @@ app.use(express.json());
...
 const PORT = 8080;
 
+app.get('/api/zipcodes/',(req,res)=>{
+
+    zipcode = req.query.zipcode;
+    distance = req.query.distance;
+    console.log(zipcode);
+    console.log(distance);
+
+    axios.get('https://www.zipcodeapi.com/rest/radius.json/'+zipcode+'/'+distance+'/km')
```
Comment (Codubee) [inline backend/index.js]: Can we use miles instead of kilometers

**Review by Codubee — CHANGES_REQUESTED** [review]
Fantastic work fixing everything! Just left one comment and then ill approve.

**Comment by Aarushi-Pandey** [comment]
Changed distance from kilometers to miles.

**Review by Codubee — APPROVED** [review]

**Pull request merged**